### ISSUE TITLE
[Discussion] Nested menu

### ISSUE THREAD

**Issue by aress31** [siriwatknp/mui-treasury#798]
Hi @siriwatknp,

Right now the `mui-layout` looks **really good** - it just need feedback from developers to get even better and cover more real-world use case as well as optimisation to avoid wasted rendering and make it more lightweight. 

It really helps me with my project and therefore I am gratefully for it, and I strongly believe that it can get even better by simplifying things. Namely, `nested-menu`, there is already this component https://mui-treasury.com/components/menu available, but it is not well documented - the API, etc. - and when I tried to use it within my project it was not working at all as if it was an empty class.

I think that more efforts should be focused on this specific component, as the `Material-UI` library does not include a similar `component`, I am thinking that maybe this component could be reworked entirely, to make it more light-weight, easy to use, well-documented and easy to style. Let's be honest there is already the `EdgeSidebar component` and most people include a `nested menu` in this component, so maybe integrating the two components more closely could be a good objective as well.

I would really love to help, but I am not a career programmer, hence a component like that ready to use with many preset styles, like the one currently used on https://mui-treasury.com/ would make my life so much easier. :)

Looking forward to hear your thoughts on that.
Alex
A

**Comment by siriwatknp** [comment]
Yeah, agree. Next plan is to improve components & styles package including documentation. #792

**Comment by aress31** [comment]
Awesome, looking forward to see the results!
